core/base/src/TString.cxx

```diff
@@ -402,6 +417,11 @@ TString& TString::Append(char c, Ssiz_t rep)
 
 Ssiz_t TString::Capacity(Ssiz_t nc)
 {
+	if (nc < 0)
```

Comment:
We don't need any special handling here.  A negative `nc` is in particular smaller than `Length()`, therefore resulting in a no-op.